main.tf

```diff
@@ -62,7 +62,8 @@ resource "aws_launch_template" "default" {
       market_type = lookup(instance_market_options.value, "market_type", null)
 
       dynamic "spot_options" {
-        for_each = flatten(list(lookup(instance_market_options.value, "spot_options", [])))
+        for_each = (instance_market_options.value.spot_options != null ?
```

Comment:
`instance_market_options.value.spot_options` - what variable or local or resource is it?

Reply:
This is from var.instance_market_options.spot_options (referenced via enclosing instance_market_options value). I changed this simply because, for some reason, the old version was giving me problems (terraform version?). AFAIKT code should mean the same thing.